modules/openideal_user/src/Plugin/Condition/UserPageCoreCondition.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+namespace Drupal\openideal_user\Plugin\Condition;
+
+use Drupal\Core\Condition\ConditionPluginBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Routing\CurrentRouteMatch;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides a 'User Page' condition.
+ *
+ * @Condition(
+ *   id = "openideal_user_page",
+ *   label = @Translation("User Page"),
+ *   context_definitions = {
+ *     "user" = @ContextDefinition("entity:user", label = @Translation("User"))
+ *   }
+ * )
+ */
+class UserPageCoreCondition extends ConditionPluginBase implements ContainerFactoryPluginInterface {
```

Comment:
I'm just wondering why there is a "Core" in the condition name?

Reply:
So in the same folder, we have a bunch of "RulesCondition" that implement the same Plugin - Condition, to identify the core conditions I put this prefix.